static declaration of 'sqlite3_snapshot_cmp' follows non-static declaration

```sh
sqlite3.c:67782:12: error: static declaration of 'sqlite3_snapshot_cmp' follows non-static declaration
static int sqlite3_snapshot_cmp(sqlite3_snapshot *p1, sqlite3_snapshot *p2){
           ^
sqlite3.c:10744:36: note: previous declaration is here
SQLITE_API SQLITE_EXPERIMENTAL int sqlite3_snapshot_cmp(
                                   ^
```

libsql-0.2.0

This looks to be a result of 519a33b9d3301c924f651198b3127746656b46a3 and in particular the change in the function signature of `sqlite3_snapshot_cmp` from `int` to `static int`. This is an api method that is injected into the amalgamation `sqlite3.c` when compile flag `-DSQLITE_SNAPSHOT_ENABLE` is turned on.

https://github.com/libsql/libsql/blob/57a49e1c2fc032ab25d86430fce6ab7ac11e90d9/src/wal.c#L4041


This can be replicated in current main with:

```
./configure --enable-all
# add -DSQLITE_SNAPSHOT_ENABLE to OPT_FEATURE_FLAGS in Makefile
make
#...
sqlite3.c:68015:12: error: static declaration of 'sqlite3_snapshot_cmp' follows non-static declaration
static int sqlite3_snapshot_cmp(sqlite3_snapshot *p1, sqlite3_snapshot *p2){
           ^
sqlite3.c:10807:36: note: previous declaration is here
SQLITE_API SQLITE_EXPERIMENTAL int sqlite3_snapshot_cmp(
                                   ^
1 error generated.
make: *** [Makefile:852: sqlite3.lo] Error 1
```

I'm going to submit a PR to revert that one line of code.

--- a/src/wal.c
+++ b/src/wal.c
@@ -4038,7 +4038,7 @@ static void sqlite3WalSnapshotOpen(
 ** Return a +ve value if snapshot p1 is newer than p2. A -ve value if
 ** p1 is older than p2 and zero if p1 and p2 are the same snapshot.
 */
-static int sqlite3_snapshot_cmp(sqlite3_snapshot *p1, sqlite3_snapshot *p2){
+int sqlite3_snapshot_cmp(sqlite3_snapshot *p1, sqlite3_snapshot *p2){
   WalIndexHdr *pHdr1 = (WalIndexHdr*)p1;
   WalIndexHdr *pHdr2 = (WalIndexHdr*)p2;
 

